src/structural_pipeline/COMMIT-Filter/commit_filter.m

```diff
@@ -0,0 +1,200 @@
+            'TEMPLATE', thisTemplate);
+        thisFilteredFiberPropertiesFile =  strrep(strrep(filteredFiberPropertiesFile, ...
+            'METHOD', thisMethod), ...
+            'TEMPLATE', thisTemplate);        
+        thisRegionPropertiesFile = strrep(regionPropertiesFile, ...
+            'TEMPLATE', thisTemplate);        
+
+        fprintf('%s - %s\n', thisTemplate, upper(thisMethod));
+
+        % Open reconstructed fiber file.
+        % This script assumes that the fibercloud data is small enough to
+        % fit into memory, which is quite a liberal assumption.
+        fprintf('Load fibers...');
+        [fibers, headerFiberFile] = readTrk(thisFiberFile);
+        fprintf(' done\n');
+
+        % Fiber properties
+        data = load(thisFiberPropertiesFile);
+        fiberProperties = data.fiberProperties;
+        propertyDescriptions = data.propertyDescriptions;
+        
+        % Filter fibers that are in connectome and group fibers into
+        % connection bundles. 
+
+        % To keep things simple we want each fiber only one time in the
+        % fiber cloud file. We will order fibers by the longest connection
+        % they contribute to.
+
+        fprintf('Filter fibers...');
+        % Sort such that first occurances of a fiber are the longest
+        [~, indx] = sort(fiberProperties(:,6), 'descend');
+        fiberProperties = fiberProperties(indx,:);
+
+        % Filter the first occurances of a fiber
+        [~, indx] = unique(fiberProperties(:,1), 'stable');
+        fiberProperties = fiberProperties(indx,:);
+
+        % Sort/group the filtered fibers by connection
+        [~, indx] = sort(fiberProperties(:,2), 'ascend');
+        fibers = fibers(fiberProperties(indx,1));
+        fprintf(' done\n');
+
+        fprintf('Write intermediate files for Python COMMIT script...');
+        voxelSize = headerFiberFile.voxel_size;
+        headerFiberFile.voxel_order = strtrim(headerFiberFile.voxel_order);
+        writeFibers(fibers, thisFilteredFiberFile, voxelSize, headerFiberFile);
+
+        % Create connectome file for COMMIT script with the updated NOS.
+        data = load(thisRegionPropertiesFile);
+        regionDescriptions = data.regionDescriptions;        
+        nRegions = length(regionDescriptions);
+        
+        assert(max(fiberProperties(:, 2)) <= 0.5*nRegions*(nRegions-1), ...
+            ['The number of regions does not match between ', ...
+            'fiberPropertiesFile (%s) and regionPropertiesFile (%s).'], ...
+            thisFiberPropertiesFile, thisRegionPropertiesFile);
+
+        W = accumarray(fiberProperties(:, 2), ones(size(fiberProperties,1),1), ...
+            [0.5*nRegions*(nRegions-1) 1]);
+        W = squareform(W);        
+        dlmwrite(intermediateConnectomeFile, W, ',');
+        fprintf(' done\n');
+
+        % Run Python script to perform COMMIT and COMMIT2
+        cmd = [pythonInterpreter ' "' commitScriptFile '"' ...
```

Comment:
We could look into [`pyrunfile`](https://nl.mathworks.com/help/matlab/ref/pyrunfile.html), although I suppose this does the job ;)

Reply:
At this moment running the python script as a system command seems most suitable because it allows the user to specify the environment and no direct interaction between python and MATLAB is necessary.